internal/host/host.go

```diff
@@ -698,6 +699,31 @@ func (m *Manager) reportInstallationMetrics(ctx context.Context, h *models.Host,
 	m.metricApi.ReportHostInstallationMetrics(log, cluster.OpenshiftVersion, h.ClusterID, cluster.EmailDomain, boot, h, previousProgress, CurrentStage)
 }
 
+func (m *Manager) ReportValidationFailedMetrics(ctx context.Context, h *models.Host) error {
+	log := logutil.FromContext(ctx, m.log)
+	var cluster common.Cluster
+	err := m.db.First(&cluster, "id = ?", h.ClusterID).Error
```

Comment:
Why take the cluster here and not get the needed parameters in the function?

Reply:
Just copied from https://github.com/ybettan/assisted-service/blob/9dddcdc75c5d53a1c1dbbc3445dd371da1bd010b/internal/host/host.go#L683-L699.

Fixing it to get it as args.